src/classes/User.js

```diff
@@ -33,63 +33,72 @@ class User {
   removeFromRecipesToCook(recipeOut) {
     this.recipesToCook.forEach(recipe => {
       if (recipeOut.name === recipe.name) {
-        this.recipesToCook.splice(recipeOut, 1)
+        this.recipesToCook.splice(this.recipesToCook.indexOf(recipeOut), 1)
       }
     })
   }
 
-  filterFavoriteRecipeTags(tags) {
-    let matchTags = [];
-    this.favoriteRecipes.forEach(recipe => {
-      tags.forEach(tag => {
-        if (recipe.tags.includes(tag)) {
-          matchTags.push(recipe);
-        }
-      })
-    })
-    if (!matchTags.length) {
-      return "No recipes match criteria."
-    } else {
-      return matchTags
-    }
+  filterFavoriteRecipeTags(inputTag) {
+    let tagged = this.favoriteRecipes.filter(recipe => recipe.tags.includes(inputTag));
+    return tagged;
   }
 
-  filterFavoriteRecipeByName(name) {
+  filterFavoriteRecipeByName(searchInput) {
     let matchName = [];
     this.favoriteRecipes.forEach(recipe => {
-      if (recipe.name.includes(name)) {
+      if (recipe.name.toLowerCase().includes(searchInput.toLowerCase())) {
         matchName.push(recipe);
+        console.log('recipe', recipe);
       }
     })
-    if (!matchName.length) {
```

Comment:
This was causing some strange behavior when the conditional was matched.  It is also unnecessary if we implement warnings for if the user's search returns zero results.